src/modules/git_status.rs

```diff
@@ -354,6 +361,107 @@ fn format_symbol(format_str: &str, config_path: &str, context: &Context) -> Opti
     format_text(format_str, config_path, context, |_variable| None)
 }
 
+#[cfg(target_os = "linux")]
+fn git_status_wsl(context: &Context, conf: &GitStatusConfig) -> Option<String> {
+    use crate::utils::create_command;
+    use nix::sys::utsname::uname;
+    use std::env;
+    use std::io::ErrorKind;
+
+    let starship_exe = conf.windows_starship?;
+
+    // Ensure this is WSL
+    // This is lowercase in WSL1 and uppercase in WSL2, just skip the first letter
+    if !uname().release().contains("icrosoft") {
+        return None;
+    }
+
+    log::trace!("Using WSL mode");
+
+    // Get Windows path
+    let winpath = match create_command("wslpath")
+        .map(|mut c| {
+            c.arg("-w").arg(&context.current_dir);
+            c
+        })
+        .and_then(|mut c| c.output())
+    {
+        Ok(r) => r,
+        Err(e) => {
+            // Not found might means this might not be WSL after all
+            let level = if e.kind() == ErrorKind::NotFound {
+                log::Level::Debug
+            } else {
+                log::Level::Error
+            };
+
+            log::log!(level, "Failed to get Windows path:\n{:?}", e);
+
+            return None;
+        }
+    };
+
+    let winpath = match std::str::from_utf8(&winpath.stdout) {
```

Comment:
Just checking: there's no way this should ever return a UNC path right?

Reply:
It does not seem to return an UNC path for long paths anyway. In any case, `dunce::canoncialize` is just `std::fs::canoncialize`, followed by `dunce::simplified` to remove the UNC prefix from `std::fs::canoncialize` if possible, so even a pre-prefixed path would be simplified.